nilearn/datasets/tests/test_atlas.py

```diff
@@ -243,6 +243,99 @@ def test_fail_fetch_atlas_harvard_oxford(tmp_path, request_mocker):
     assert ho.labels[6] == "Right R3"
 
 
+def test_fail_fetch_atlas_juelich(tmp_path, request_mocker):
+    # specify non-existing atlas item
+    with pytest.raises(ValueError, match='Invalid atlas name'):
+        atlas.fetch_atlas_juelich('not_inside')
+
+    # specify existing atlas item
+    target_atlas = 'maxprob-thr0-1mm'
+    target_atlas_fname = 'Juelich-' + target_atlas + '.nii.gz'
+    ho_dir = str(tmp_path / 'fsl' / 'data' / 'atlases')
+    os.makedirs(ho_dir)
+    nifti_dir = os.path.join(ho_dir, 'Juelich')
+    os.makedirs(nifti_dir)
+
+    target_atlas_nii = os.path.join(nifti_dir, target_atlas_fname)
+
+    # Create false atlas
+    atlas_data = np.zeros((10, 10, 10), dtype=int)
+
+    # Create interhemispheric maps
+    atlas_data[:, :2, :] = 1
+
+    atlas_data[:5, 3:5, :] = 2
+    atlas_data[5:, 3:5, :] = 2
+
+    atlas_data[:5, 7:9, :] = 3
+    atlas_data[5:, 7:9, :] = 3
+
+
+    nibabel.Nifti1Image(atlas_data, np.eye(4) * 3).to_filename(
+        target_atlas_nii)
+
+    dummy = open(os.path.join(ho_dir, 'Juelich.xml'), 'w')
```

Comment:
please rely on context managers to release resources such as open file handles:
```
with open("...Juelich.xml") as dummy:
    dummy.write("...")
# dummy is closed automatically
```
or:
```
pathlib.Path("...Juelich.xml").write_text("...")
```